recipes/sqlite3/all/conanfile.py

```diff
@@ -167,6 +170,10 @@ def _create_cmake_module_variables(module_file):
     def _module_file_rel_path(self):
         return os.path.join("lib", "cmake", "conan-official-{}-variables.cmake".format(self.name))
 
+    def requirements(self):
+        if self.options.enable_icu:
+            self.requires("icu/70.1")
+
```

Comment:
Could you move this just before `validate()` please? It's easier to understand a recipe when methods are ordered by conan flow of execution.